src/lib/widget/_utilities/api.utilities.ts

```diff
@@ -1,254 +1,213 @@
-              journey,
-              oauth,
-              user: response,
-            });
-          if (modal) {
-            modal.close({ reason: 'auto' });
+        } else if ($journeyStore.successful) {
+          if (requestsOauth && $oauthStore.loading === false && $oauthStore.completed === false) {
+            oauthStore.get();
+          } else if (!requestsOauth) {
+            modal && modal.close({ reason: 'auto' });
           }
-        } else if (response.error) {
-          returnError &&
-            returnError({
-              journey,
-              oauth,
-              user: response,
-            });
         }
-        /**
-         * Clean up unneeded subscription, but only when it's successful
-         * Leaving the subscription allows for the journey to be
-         * restarted internally.
-         */
-        if (response.successful) {
-          userStoreUnsub();
-        }
-      });
+      },
+    );
 
-      if (options?.resumeUrl) {
-        journeyStore.resume(options.resumeUrl);
+    function start(startOptions?: JourneyOptionsStart) {
+      if (startOptions?.resumeUrl) {
+        journeyStore.resume(startOptions.resumeUrl);
       } else {
         journeyStore.start({
-          ...options?.config,
-          tree: options?.journey,
+          ...startOptions?.config,
+          tree: startOptions?.journey,
         });
       }
-    },
-    onFailure(fn: (response: Response) => void) {
-      returnError = (response) => fn(response);
-    },
-    onSuccess(fn: (response: Response) => void) {
-      returnResponse = (response) => fn(response);
-    },
+      return new Promise((resolve) => {
+        const unsubscribe = journeyStore.subscribe((event) => {
+          if (event.completed) {
+            resolve(event);
+            unsubscribe();
+          }
+        });
+      });
+    }
+
+    return { start, subscribe };
   };
   const user = {
-    async authorized(remote = false) {
-      if (remote) {
-        try {
-          return await UserManager.getCurrentUser();
-        } catch (err) {
-          console.warn(err);
-          return;
-        }
-      }
-      return !!(await TokenManager.getTokens());
-    },
-    async info(remote = false) {
-      userStore = userStore as UserStore;
-      if (remote) {
-        try {
-          return await UserManager.getCurrentUser();
-        } catch (err) {
-          console.warn(err);
-          return;
-        }
+    info() {
+      const { get, subscribe } = userStore;
+
+      function wrappedGet(options: ConfigOptions) {
+        get(options);
+        return new Promise((resolve) => {
+          const unsubscribe = userStore.subscribe((event) => {
+            if (event.completed) {
+              resolve(event);
+              unsubscribe();
+            }
+          });
+        });
       }
-      return get(userStore).response;
+
+      return { get: wrappedGet, subscribe };
     },
-    logout: async () => {
+    async logout() {
```

Comment:
Here's the biggest exception. Since logout is just an action and doesn't return any data or state, I have it as an async function. I _could_ have this as a method returned in one of the other APIs, user or oauth, but it's not really one or the other, but it affects both. So, for now, it's its own thing.